Fix documentation for MMDS.

Khalian opened the pull request:
The commands for updating MMDS were incorrect. Fixing them.

## Reason for This PR

Fixing documentation. 
## Description of Changes

The commands for MMDS updates were incorrect. There were " around the data field on the MMDS update commands. That is a syntax error. It should be '. Fixing that. 
 

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

andrew-d commented on docs/mmds/mmds-user-guide.md:
````diff
@@ -4,6 +4,17 @@ The Firecracker microVM Metadata Service (MMDS) is a mutable data store which
+Set your firecracker unix socket here (replace it with the absolute path to where the socket is located). Example
+
+```
+UNIX=SOCKET=/tmp/firecracker.socket
````
```suggestion
UNIX_SOCKET=/tmp/firecracker.socket
```

Khalian replied:
whats the difference here?

andrew-d commented on docs/mmds/mmds-user-guide.md:
````diff
@@ -17,7 +28,7 @@ interface configuration API can be found in the
 ```bash
 MMDS_NET_IF=eth0
-curl --unix-socket /tmp/firecracker.socket -i                 \
+curl --unix-socket $UNIX=SOCKET -i                 \
````
```suggestion
curl --unix-socket $UNIX_SOCKET -i                 \
```

andrew-d reviewed (commented):
Drive-by review (sorry!), but noticed the mistaken `=` here.

iulianbarbu commented on docs/mmds/mmds-user-guide.md:
```diff
@@ -4,6 +4,17 @@ The Firecracker microVM Metadata Service (MMDS) is a mutable data store which
+## Setup Firecracker socket
+
+
+Set your firecracker unix socket here (replace it with the absolute path to where the socket is located). Example
```
```suggestion
Set your Firecracker unix domain socket path here (replace it with the absolute path to where the socket is located). Example:
```

iulianbarbu commented on docs/mmds/mmds-user-guide.md:
````diff
@@ -17,7 +28,7 @@ interface configuration API can be found in the
 ```bash
 MMDS_NET_IF=eth0
-curl --unix-socket /tmp/firecracker.socket -i                 \
+curl --unix-socket $<span class="x x-first x-last">UNIX_SOCKET</span> -i                 \
````
What is the use of the span tag? It appears as literal text within the .md, so it has not effect.

Khalian replied:
some other reviewer suggested this to me.

Khalian replied:
reverted.

iulianbarbu reviewed (commented):
Please explicitly sign the commit with: `git commit -s --amend`.
Also, can you rephrase the commit title & description with respect to 50/72?

E.g. title: `mmds: added setup regarding to Firecracker UDS path`.
E.g. description: `MMDS user guide used a hardcoded assumed UDS path for the Firecracker API server. Replaced the hardcoded path with an env variable which should be configured before executing any API server request.`

Khalian commented:
> Please explicitly sign the commit with: `git commit -s --amend`.
> Also, can you rephrase the commit title & description with respect to 50/72?
> 
> E.g. title: `mmds: added setup regarding Firecracker UDS path`.
> E.g. description: `MMDS user guide used a hardcoded assumed UDS path for the communication with the Firecracker API server. Replaced the hardcoded path with an env variable which should be configured before executing any API server request.`

the problem I was trying to fix was not even that. It was the quotes on the json data " vs '. How do you want me to phrase this ?

iulianbarbu commented:
> The whole UNIX_SOCKET part was basically me just improving shell usage.

Indeed.

> The problem I was trying to fix was not even that. It was the quotes on the json data " vs '. How do you want me to phrase this ?

Thinking more on it, these two changes fit better on two separated commits: one for invalid syntax and one for UDS path setup. I think the first one can simply refer to the JSON invalid syntax for the requests payloads from MMDS user guide and the second one to the UDS path setup improvement. Is this more clear now?

Khalian commented:
> > The whole UNIX_SOCKET part was basically me just improving shell usage.
> 
> Indeed.
> 
> > The problem I was trying to fix was not even that. It was the quotes on the json data " vs '. How do you want me to phrase this ?
> 
> Thinking more on it, these two changes fit better on two separated commits: one for invalid syntax and one for UDS path setup. I think the first one can simply refer to the JSON invalid syntax for the requests payloads from MMDS user guide and the second one to the UDS path setup improvement. Is this more clear now?

I think thats fair. The thing that actually needs to be fixed is the quotes. The whole shell var is a quality of life improvement.

iulianbarbu commented on docs/mmds/mmds-user-guide.md:
```diff
@@ -21,8 +21,8 @@ curl --unix-socket /tmp/firecracker.socket -i                 \
   -H 'Content-Type: application/json'                         \
-  -d '{
       "iface_id": "${MMDS_NET_IF}",
+  -d '{
```
This is misplaced. It should contain `iface_id` field as well.

Khalian replied:
Oops. Fixed. Bad vim foo there.

iulianbarbu reviewed (commented):
@Khalian , besides a small change left in the review, please adjust the commit title to 50 chars. We usually do commits with respect to 50/72 rule.

Khalian commented:
> @Khalian , besides a small change left in the review, please adjust the commit title to 50 chars. We usually do commits with respect to 50/72 rule.

Uh ok. Made it more succinct.

iulianbarbu reviewed (approved):
LGTM! Thanks.

Khalian commented:
Anything else that needs to be done?
